guide/event-handling/README.md

```diff
@@ -0,0 +1,284 @@
+
+```diff
+for (const file of eventFiles) {
+	const event = require(`./events/${file}`);
+	if (event.once) {
+-		client.once(event.name, (...args) => event.execute(...args));
++		client.once(event.name, (...args) => event.execute(...args, client));
+	} else {
+-		client.on(event.name, (...args) => event.execute(...args));
++		client.on(event.name, (...args) => event.execute(...args, client));
+	}
+}
+```
+
+Now, `client` will be available as the last argument to the `execute` function in every event file you'll create. Since, `ready` event doesn't return anything, `client` will be the only argument to the `execute` function of `ready.js` event file. You can make use of `client` in `ready.js` by logging your bot's name in the console when it becomes ready. Head back to `ready.js` and make the following changes to it:
+
+```diff
+module.exports = {
+	name: 'ready',
+    once: true,
++	execute(client) {
+-		console.log('Ready!');
++		console.log(`Logged in as ${client.user.tag}`);
+	},
+};
+```
+
+You can omit the `client` argument from the `execute` function in an event file if you either obtain it from some other discord.js structure or don't need the `client` in it at all. For example, the `message.js` event file will still work if you don't take `client` as an argument in the `execute` function, but if you want you can make the following changes, and it will work just like earlier: 
+
+```diff
+module.exports = {
+	name: 'message',
+	once: false,
++	execute(message, client) {
+		if (!message.content.startsWith(prefix) || message.author.bot) return;
+-		const client = message.client;
+		// ...
+	},
+};
+```
+
+::: warning
+Always, make sure that you take all the valid arguments returned by an event first and then take the `client` as the last argument.
+:::
+
+It is worth to note that the position of `client` argument matters. For example `messageUpdate` event returns `oldMessage` and `newMessage` objects. Your event file for this event should look something like this:
+
+```js
+// This will work
+module.exports = {
+	name: 'messageUpdate',
+	once: false,
+	execute(oldMessage, newMessage, client) {
+		// ...
+	},
+};
+```
+
+If you aren't using `client` in an event file, you can omit it from the function arguments, and it will still work without any errors:
+
+```js
+// This will work too
+module.exports = {
+	name: 'messageUpdate',
+	once: false,
+	execute(oldMessage, newMessage) {
+		// ...
+	},
+};
+```
+
+What you cannot do is pass the `client` argument at the wrong position, like this:
+
+```js
+// This will create problems :(
+module.exports = {
+	name: 'messageUpdate',
+	once: false,
+	execute(newMessage, client) {
+		// ...
+	},
+};
+```
```

Comment:
```suggestion
It is worth noting that the position of `client` argument matters. For example, the `messageUpdate` event has two arguments: `oldMessage` and `newMessage`. Your event file for this event would look something like this:

```js
module.exports = {
	name: 'messageUpdate',
	execute(oldMessage, newMessage, client) {
		// ...
	},
};
```

If you were to try `execute(newMessage, client)`, this would mean that `newMessage` is an old `Message` object and `client` is a new `Message` object.
```